examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -493,7 +493,8 @@ CHIP_ERROR CastingServer::VerifyOrEstablishConnection(TargetVideoPlayerInfo & ta
         prevDeviceProxy->Disconnect();
     }
 
-    CastingServer::GetInstance()->mActiveTargetVideoPlayerInfo = targetVideoPlayerInfo;
+    int cacheIndex                                             = GetVideoPlayerIndex(&targetVideoPlayerInfo);
+    CastingServer::GetInstance()->mActiveTargetVideoPlayerInfo = mCachedTargetVideoPlayerInfo[cacheIndex];
     uint32_t delay                                             = 0;
     if (targetVideoPlayerInfo.IsAsleep())
```

Comment:
don't we need to stop using `targetVideoPlayerInfo` altogether if we think there is memory access issues here?

Reply:
The issue here specifically has to do with object lifetimes.

The caller of CastingServer.VerifyOrEstablishConnection() allocates the object on its stack. See here: [https://github.com/sharadb-amazon/connectedhomeip/blob/v1.0.0-branch/examples/tv-casting-app/android/App/app/src/main/jni/cpp/TvCastingApp-JNI.cpp#L261](https://github.com/sharadb-amazon/connectedhomeip/blob/v1.0.0-branch/examples/tv-casting-app/android/App/app/src/main/jni/cpp/TvCastingApp-JNI.cpp#L261)

But the API uses the = operator here which only does a shallow copy of the passed in targetVideoPlayerInfo (which contains char arrays, pointers, other objects with pointers). After the API call is complete, the stack unwinds and the targetVideoPlayerInfo parameter goes out of scope / gets deallocated. At this point pointers in mActiveTargetVideoPlayerInfo may be pointing to freed memory spaces.

To fix this problem, we find the same targetVideoPlayerInfo in the mCachedTargetVideoPlayerInfo that are maintained through the life of the CastingServer, and copy the values from there.